src/dune_rules/coq_rules.ml

```diff
@@ -347,7 +342,17 @@ let generic_coq_args ~sctx ~dir ~wrapper_name ~boot_type ~mode ~coq_prog
     ~stanza_flags ~ml_flags ~theories_deps ~theory_dirs coq_module =
   let+ coq_stanza_flags =
     let+ expander = Super_context.expander sctx ~dir in
-    let coq_flags = coq_flags ~expander ~dir ~stanza_flags ~sctx in
+    let coq_flags =
+      let open Action_builder.O in
+      let* standard = Action_builder.of_memo @@ Super_context.coq ~dir sctx in
+      let standard =
+        (* For dune coq top, we don't want the -q flag *)
+        match coq_prog with
+        | `Coqtop -> Action_builder.return []
+        | _ -> standard
```

Comment:
That seems wrong, IIUC then coqtop will ignore the flags set in the workspace?

Reply:
Yes, this is not ideal. We could explicitly do `:standard \ -q` here like you suggested.

Reply:
Yes, that should work for now.

Reply:
Actually this doesn't remove the stanza flags since we have a test for that. What it does remove are any extra flags set via env.

Reply:
It also appears as though the env stanza is broken for coq flags. I might be inclined to just remove that feature completely as it doesn't work today.

Reply:
How come it is broken? I added pretty exhaustive test-suite for it.

Reply:
When I add an env with coq flags to coqtop-flags, I see that both coqc and coqtop do not inherit the flags.

Reply:
Anyway that is a separate issue. I've moved the "default" flag -q into coq_rules, and left the env node empty.